src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTests.cs

```diff
@@ -16692,5 +16692,36 @@ static void Main()
 none called
 in called");
         }
+
+        [Fact]
+        public void NormalizedNaN()
+        {
+            string source =
+@"using System;
+class Program
+{
+    static void Main()
+    {
+        CheckNaN(double.NaN);
+        CheckNaN(float.NaN);
+        CheckNaN(0.0 / 0.0);
+        CheckNaN(0.0 / -0.0);
```

Comment:
Is testing `-0.0 / 0.0` and `-0.0 / -0.0` also important?

Reply:
There are a huge number of ways of producing NaN values. I'm just trying a small representative sample of them.